crates/runtime/src/accelerated_table/retention.rs

```diff
@@ -102,15 +105,23 @@ impl super::AcceleratedTable {
                     }
                 }
 
-                tracing::trace!("[retention] Exprs {exprs:?}");
+                // Combine all expressions into a single OR expression as time and SQL expressions are applied independently
+                let Some(expr) = exprs.into_iter().reduce(Expr::or) else {
+                    tracing::warn!(
+                        "[retention] No valid retention filters found for dataset {dataset_name}"
+                    );
+                    continue;
+                };
+
+                tracing::trace!("[retention] Expr {expr:?}");
 
                 let ctx = SessionContext::new_with_config_rt(
                     get_df_default_config(),
                     default_runtime_env(),
                 );
 
                 let plan = deleted_table_provider
-                    .delete_from(&ctx.state(), &exprs)
+                    .delete_from(&ctx.state(), &[expr])
```

Comment:
Just to learn, the filters in `DeletionTableProvider` are supposed to be used as an AND  (i.e. only delete data that return true for all binary expression in `&[Expr]`)? 
```rust
#[async_trait]
pub trait DeletionTableProvider: TableProvider {
    async fn delete_from(
        &self,
        _state: &dyn Session,
        _filters: &[Expr],
    ) -> datafusion::error::Result<Arc<dyn ExecutionPlan>>;
}
```

Reply:
@Jeadie  - this is correct, thank you for review
> filters in DeletionTableProvider are supposed to be used as an AND (i.e. only delete data that return true for all binary expression in &[Expr])